afar/core.py

```diff
@@ -124,7 +125,37 @@ def __enter__(self):
             if self.data:
                 raise RuntimeError("uh oh!")
             self.data = {}
-        lines, offset = inspect.findsource(self._frame)
+        try:
+            lines, offset = inspect.findsource(self._frame)
+        except OSError:
+            # Try to fine the source if we are in %%time or %%timeit magic
+            if (
+                self._frame.f_code.co_filename in {"<timed exec>", "<magic-timeit>"}
+                and reprs.in_ipython()
+            ):
+                import IPython
+
+                ip = IPython.get_ipython()
+                cell = ip.history_manager._i00  # The current cell!
+                lines = cell.splitlines(keepends=True)
+                # strip the magic
+                for i, line in enumerate(lines):
+                    if line.strip().startswith("%%"):
```

Comment:
I don't imagine there would be other lines that start with `%%`, but since we're only supporting `%%time` and `%%timeit` anyways, I don't think this would hurt

```suggestion
                    if line.strip().startswith("%%time"):
```